core/txpool/blobpool/blobpool.go

```diff
@@ -873,6 +889,92 @@ func (p *BlobPool) Reset(oldHead, newHead *types.Header) {
 	p.updateStorageMetrics()
 }
 
+// convertSidecars runs on a separate thread and manages the pool lock as needed
+// during the conversion. Any txs that fail to convert are dropped.
+func (p *BlobPool) convertSidecars(txs []*blobTxMeta, cancel chan struct{}) {
+	convert := func(m *blobTxMeta) error {
+		if m.version != types.BlobSidecarVersion0 {
+			return nil
+		}
+		p.lock.RLock()
+		data, err := p.store.Get(m.id)
+		if err != nil {
+			// This will most likely occur when a tx gets included before it can be
+			// converted locally.
+			p.lock.RUnlock()
+			return fmt.Errorf("unable to load tx for conversion: %w", err)
+		}
+		p.lock.RUnlock()
+
+		tx := new(types.Transaction)
+		if err := rlp.DecodeBytes(data, tx); err != nil {
+			return fmt.Errorf("failed to decode stored blob transaction: %w", err)
+		}
+		// Perform conversion.
+		sidecar := tx.BlobTxSidecar()
+		if sidecar == nil {
+			return fmt.Errorf("missing sidecar")
+		}
+		if err := sidecar.ToV1(); err != nil {
+			return fmt.Errorf("failed to convert sidecar: %w", err)
+		}
+
+		// Reencode and insert into store.
+		blob, err := rlp.EncodeToBytes(tx)
+		if err != nil {
+			return fmt.Errorf("failed to encode new sidecar: %w", err)
+		}
+		p.lock.Lock()
+		defer p.lock.Unlock()
+		id, err := p.store.Put(blob)
+		if err != nil {
+			return fmt.Errorf("failed to write tx with converted sidecar: %w", err)
+		}
+
+		// Update metadata and trackers for the tx.
+		m.id = id
```

Comment:
You also have to update the `size` and `storageSize` in the blobMetadata

Reply:
It's the main reason I go with #32665, there are a tons of internal fields in the blobPool, and it's very easy to break them.